apps/web/src/pages/PoolCandidates/ViewPoolCandidatePage/components/MoreActions/FinalDecisionDialog.tsx

```diff
@@ -0,0 +1,339 @@
+        defaultMessage: "Error: could not update pool candidate status",
+        id: "FSlrKF",
+        description:
+          "Message displayed when an error occurs while an admin updates a pool candidate",
+      }),
+    );
+  };
+
+  const handleFormSubmit: SubmitHandler<FormValues> = async (
+    values: FormValues,
+  ) => {
+    let computedStatus = poolCandidateStatus;
+    if (values.finalAssessmentDecision === "qualified") {
+      computedStatus = PoolCandidateStatus.QualifiedAvailable;
+    }
+    if (
+      values.finalAssessmentDecision === "disqualified" &&
+      values.disqualifiedDecision === "assessment"
+    ) {
+      computedStatus = PoolCandidateStatus.ScreenedOutAssessment;
+    }
+    if (
+      values.finalAssessmentDecision === "disqualified" &&
+      values.disqualifiedDecision === "application"
+    ) {
+      computedStatus = PoolCandidateStatus.ScreenedOutApplication;
+    }
+
+    const statusObject = { status: computedStatus };
+    const massagedValues = values.expiryDate
+      ? { ...statusObject, expiryDate: values.expiryDate }
+      : statusObject;
+
+    await executeMutation({ id: poolCandidateId, input: massagedValues })
+      .then((result) => {
+        if (result.data?.updatePoolCandidateAsAdmin) {
+          toast.success(
+            intl.formatMessage({
+              defaultMessage: "Pool candidate status updated successfully",
+              id: "uSdcX4",
+              description:
+                "Message displayed when a pool candidate has been updated by and admin",
+            }),
+          );
+          setIsOpen(false);
+        } else {
+          handleError();
+        }
+      })
+      .catch(() => {
+        handleError();
+      });
+  };
+
+  return (
+    <Dialog.Root open={isOpen} onOpenChange={setIsOpen}>
+      <Dialog.Trigger>
+        <Button type="button" color="primary" mode="solid">
+          {intl.formatMessage({
+            defaultMessage: "Record final decision",
+            id: "ngHHmI",
+            description: "Button to click on to fill out a final decision form",
+          })}
+        </Button>
+      </Dialog.Trigger>
+      <Dialog.Content>
+        <Dialog.Header>
+          {intl.formatMessage({
+            defaultMessage: "Final assessment decision",
+            id: "qqkQ/h",
+            description: "Final assessment decision dialog header",
+          })}
+        </Dialog.Header>
+        <Dialog.Body>
+          <p>
+            {intl.formatMessage({
+              defaultMessage:
+                "Use this section to finalize the decision on this candidate's application after you've performed all necessary assessments.",
+              id: "mF9apF",
+              description: "Text describing a dialog's purpose",
+            })}
+          </p>
+          <Heading
+            data-h2-font-weight="base(700)"
+            data-h2-margin-bottom="base(x.5)"
+            data-h2-font-size="base(normal)"
+          >
```

Comment:
This heading should be `level="h3" size="h6"`. This should avoid the need for the extra styles and provide the proper hierarchy.

Reply:
Done in https://github.com/GCTC-NTGC/gc-digital-talent/pull/8978/commits/019d9ab14ea31a6398db07bd9cc0566e8860d8bd